contracts/mainnet/connectors/liquity/main.sol

```diff
@@ -50,23 +29,23 @@ abstract contract LiquityResolver is Events, Helpers {
      * @param borrowAmount The amount of LUSD to borrow
      * @param upperHint Address of the Trove near the upper bound of where the user's Trove should now sit in the ordered Trove list
      * @param lowerHint Address of the Trove near the lower bound of where the user's Trove should now sit in the ordered Trove list
-     * @param getId Optional storage slot to retrieve ETH from
-     * @param setId Optional storage slot to store the LUSD borrowed against
+     * @param getIds Optional (default: 0) Optional storage slot to get deposit & borrow amounts stored using other spells
+     * @param setIds Optional (default: 0) Optional storage slot to set deposit & borrow amounts to be used in future spells
     */
     function open(
         uint depositAmount,
         uint maxFeePercentage,
         uint borrowAmount,
         address upperHint,
         address lowerHint,
-        uint getId,
-        uint setId
+        uint[] getIds,
+        uint[] setIds
     ) external payable returns (string memory _eventName, bytes memory _eventParam) {
-        if (getId != 0 && depositAmount != 0) {
-            revert("open(): Cannot supply a depositAmount if a non-zero getId is supplied");
-        }
 
-        depositAmount = getUint(getId, depositAmount);
+        depositAmount = getUint(getIds[0], depositAmount);
+        borrowAmount = getUint(getIds[1], borrowAmount);
```

Comment:
Is the purpose of making the `borrowAmount` gettable to enable other spells to control amount that gets borrowed? What would some examples be where the `borrowAmount` needs to be gettable?

Reply:
In refinancing where you need to do 100% refinancing. Eg:- let's say LUSD is supported by Aave now their could be Aave & Liquity refinancing. When a user wants to move 100% debt, he'll borrow flashloan & payback 100% on Aave using `uint(-1)` and store it in an ID then use that ID to borrow the same amount from Liquity.

Also, in general, we prefer to keep things open even if we can't think of any use cases because it has happened with us before where we found a new use case and then have to deploy newly updated connector for it.